test/integration/interactive_test.js

```diff
@@ -0,0 +1,149 @@
+var assert = require('assert');
+var base = require('taskcluster-base');
+var cmd = require('./helper/cmd');
+var crypto = require('crypto');
+var debug = require('debug')('docker-worker:test:interactive-test')
+var DockerExecClient = require('docker-exec-websocket-server').DockerExecClient;
+var DockerWorker = require('../dockerworker');
+var TestWorker = require('../testworker');
+var Promise = require('promise');
+var request = require('superagent-promise');
+var slugid = require('slugid');
+
+suite('use docker exec websocket server', () => {
+  let worker;
+  setup(async () => {
+    worker = new TestWorker(DockerWorker);
+    await worker.launch();
+  });
+
+  teardown(async () => {
+    if (worker) {
+      await worker.terminate();
+      worker = null;
+    }
+  });
+
+  async function getWithoutRedirect (url) {
+    var res = await request.get(url).redirects(0).end();
+    return res.headers.location;
+  };
+
+  test('cat', async () => {
+    let taskId = slugid.v4();
+  	let task = {
+      payload: {
+        image: 'busybox',
+        command: cmd('sleep 15'),
+        maxRunTime: 2 * 60,
+        features: {
+          interactive: true
+        }
+      }
+    };
+    debug('posting to queue');
+    worker.postToQueue(task, taskId);
+    
+    var passed = false;
+
+    var signedUrl = worker.queue.buildSignedUrl(
+        worker.queue.getLatestArtifact,
+        taskId,
+        'private/mozilla/interactive.sock',
+        {expiration: 60 * 5});
+
+    var url;
+    await base.testing.poll(async () => {
+      url = await getWithoutRedirect(signedUrl);
+      assert(url, 'artifact not found');
+    }, 20, 1000);
+
+    //for testing, we don't care about https verification
+    var client = new DockerExecClient({
+      tty: false,
+      command: ['cat'],
+      url: url,
+      wsopts: {rejectUnauthorized: false}
```

Comment:
neat, hack I like that... people will rarely need to provide options for `ws` anyways :)